### ISSUE TITLE
[feature] add follow symbolic links option to CLI

### ISSUE THREAD

**Issue by john-nils** [ast-grep/ast-grep#1461]
As I'm attempting to integrate ast-grep into bazel I noticed that ast-grep seems to skip symbolic links when scanning (bazel uses symbolic links a lot).

Can be reproduced by.
```
$ mkdir tmp
$ cd tmp
$ echo 'void fn() { assert(0); }' > test.cpp
$ ln -s test.cpp symlink_test.cpp
$ ast-grep run --pattern 'assert($$$ARGS)'
# ast-grep scan has similar problem, but is not a one-liner like above
```

# 🙁 Actual behavior

```
test.cpp
1│void fn() { assert(0); }
```

# 🙂 Expected behavior

```
test.cpp
1│void fn() { assert(0); }

symlink_test.cpp
1│void fn() { assert(0); }
```

**Comment by HerringtonDarkholme** [comment]
ast-grep will follow ripgrep's decision https://github.com/BurntSushi/ripgrep/issues/692

**Comment by john-nils** [comment]
Thank you for clarification. And agree - just didn't think about that aspect.
I was able to workaround it by giving all files explicitly on command-line (avoiding dir-walk).
For rule-files that are softlinked i updated sgconfig.yml to point to each individual file under "ruleDirs". That also worked fine.

Thank you for the excellent tool - it's a great light-weight linter to complement clang-tidy.

**Comment by HerringtonDarkholme** [comment]
Will ship this change in upcoming weekend!

**Comment by HerringtonDarkholme** [comment]
Published in 0.27.2!
